### ISSUE TITLE
CA1054 fires when implementing an interface method

### ISSUE THREAD

**Issue by alexrp** [dotnet/roslyn-analyzers#6371]
### Analyzer

**Diagnostic ID**: [CA1054](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1054): `URI parameters should not be strings`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.101](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

CA1054 fires when implementing an interface method. In this case, there is nothing I can do about the fact that the parameter is typed as `string` instead of `Uri`.

### Steps To Reproduce

Using the [SendGrid](https://www.nuget.org/packages/SendGrid) API:

```csharp
public sealed class MySendGridClient : ISendGridClient
{
    // ... snip ...

    public Task<Response> RequestAsync(
        BaseClient.Method method,
        string? requestBody = null,
        string? queryParams = null,
        string? urlPath = null, // <- CA1054
        CancellationToken cancellationToken = default)
    {
        throw new NotImplementedException();
    }

    // ... snip ...
}
```

### Expected behavior

CA1054 should not fire in this case as it is not actionable.

### Actual behavior

CA1054 fires on the `urlPath` parameter.
